File: ideal-core/src/main/java/cn/idealframework2/cache/memory/CaffeineMemoryCache.java
package cn.idealframework2.cache.memory;

import javax.annotation.Nonnull;
import javax.annotation.Nullable;
import java.util.Map;
import java.util.function.Function;

/**
 * @author 宋志宗 on 2022/12/5
 */
public class CaffeineMemoryCache<K, V> implements MemoryCache<K, V> {

  @Nullable
  @Override
  public V getIfPresent(@Nonnull K key) {
    return null;
  }

  @Nullable
  @Override
  public V get(@Nonnull K key, @Nonnull Function<K, V> function) {
    return null;
  }

  @Override
  public void put(@Nonnull K key, @Nonnull V value) {

  }

  @Override
  public void putAll(@Nonnull Map<K, V> map) {

  }

  @Override
  public void invalidate(@Nonnull K key) {

  }

  @Override
  public void invalidateAll(@Nonnull Iterable<K> keys) {

  }
}

File: ideal-core/src/main/kotlin/cn/idealframework2/cache/coroutine/memory/CaffeineMemoryCache.kt
package cn.idealframework2.cache.coroutine.memory

import cn.idealframework2.cache.memory.MemoryCache
import java.util.function.Function

/**
 * @author 宋志宗 on 2022/12/5
 */
class CaffeineMemoryCache<K : Any, V : Any> : MemoryCache<K, V> {

  override fun getIfPresent(key: K): V? {
    TODO("Not yet implemented")
  }

  override fun get(key: K, function: Function<K, V>): V? {
    TODO("Not yet implemented")
  }

  override fun put(key: K, value: V) {
    TODO("Not yet implemented")
  }

  override fun putAll(map: MutableMap<K, V>) {
    TODO("Not yet implemented")
  }

  override fun invalidate(key: K) {
    TODO("Not yet implemented")
  }

  override fun invalidateAll(keys: MutableIterable<K>) {
    TODO("Not yet implemented")
  }
}
